docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,448 @@
+
+### Goals
+
+- To design a solution for orchestrating Cluster add-ons.
+- To leverage existing package management tools such as Helm for all the foundational capabilities of add-on management, i.e. add-on packages/repository, templating/configuration, add-on creation, upgrade and deletion etc. 
+- To make add-on management in Cluster API modular and pluggable, and to make it simple for developers to build a Cluster API Add-on Provider based on any package management tool, just like with infrastructure and bootstrap providers.
+
+### Non goals
+
+- To implement a full fledged package management tool in Cluster API; there are already several awesome package management tools in the ecosystem, and CAPI should not reinvent the wheel. 
+- To provide a mechanism for altering, customizing, or dealing with single Kubernetes resources defining a Cluster add-on, i.e. Deployments, Services, ServiceAccounts. Cluster API should treat add-ons as opaque components and delegate all the operations impacting add-on internals to the package management tool.
+- To expect users to use a specific package management tool.
+- To implement a solution for installing add-ons on the management cluster itself.
+
+### Future work
+
+- Handle the upgrade of Cluster add-ons when the workload Cluster is upgraded to a new Kubernetes version.
+- Provide support for advanced orchestration use cases leveraging on Cluster API’s lifecycle hooks for Cluster API.
+- Introduce capabilities to reconcile add-ons in an order that respects dependencies between add-ons.
+- Deprecate or remove the ClusterResourceSet experimental feature.
+
+## Proposal
+
+This document introduces the concept of a ClusterAddonProvider, a new pluggable component to be deployed on the Management Cluster. A ClusterAddonProvider must be implemented using a specified package management tool and will act as an intermediary/proxy between Cluster API and the chosen package management solution.
+
+This proposal also introduces a ClusterAddonProvider for [Helm](https://helm.sh/) as a concrete example of how a ClusterAddonProvider would work with a package management tool. Users can configure ClusterAddonProvider for Helm through a new Kubernetes CRD called HelmChartProxy to be deployed on the Management Cluster. HelmChartProxy will act as an intermediary (proxy) between Cluster API and the Helm to install packages onto workload clusters. The implementation of ClusterAddonProvider for Helm can be used as a reference to implement ClusterAddonProviders for other package managers, i.e. ClusterAddonProvider for kapp.
+
+### User Stories
+
+#### Story 1
+
+As a user/operator, I would like Cluster add-on providers to automatically include or install the package management tool.
```

Comment:
user is an ambiguous term, It'd be clear to refer to specific personas e.g cluster operator, dev...